doc/api/assert.md

```diff
@@ -321,10 +321,16 @@ added: REPLACEME
 * `message` {any}
 
 Awaits for the promise returned by function `block` to complete and not be
-rejected. See [`assert.rejects()`][] for more details.
+rejected.
+
+Please note: Using `assert.doesNotReject()` is actually not useful because there
+is little benefit by catching a rejection and then rejecting it again. Instead,
+consider adding a comment next to the specific code path that should not reject
+and keep error messages as expressive as possible.
```

Comment:
Suggestions: Get rid of `Please note:` and keep the first sentence concise:

```markdown
There is little benefit to using `assert.doesNotReject()`. Instead, consider adding a comment...
```

Reply:
In this specific case I would rather keep the `Please note` as I really want users to read this. That is also why it is placed prominently.